### PR TITLE
Run Fixing Algorithms when opening splits

### LINKED ISSUE #21: Run Fixing Algorithms when opening splits

### PR DESCRIPTION
**Pull request by TheEssenceOfDarkness**
closes #21

edit: I'm probably missing something here. All of the tests are still passing on my machine though.

### CODE REVIEW

**Comment by CryZe** [comment]
You may not have run `cargo update` for a while, so some dependencies are slightly outdated, which is why the tests still work for you locally. You may want to rebase your branch onto the latest master anyway, where we use the perceptual hashing now.

Also you should probably add a test here where we check that the Run does indeed get fixed when creating a Timer / Run Editor / ...

File: src/timing/timer/tests/mod.rs [hunk 1]
```diff
@@ -617,3 +617,51 @@ fn reset_and_set_attempt_as_pb() {
...
+    let run = timer.into_run(true);
+
+    let segments = run.segments();
+
+    assert_eq!(segments[0].segment_history().try_get_min_index(), Some(0));
+    assert_eq!(segments[0].segment_history().try_get_max_index(), Some(1));
+
+    assert_eq!(segments[1].segment_history().try_get_min_index(), Some(0));
+    assert_eq!(segments[1].segment_history().try_get_max_index(), Some(1));
+}
```
Comment (TheEssenceOfDarkness) [inline src/timing/timer/tests/mod.rs]: These tests will not compile due to run.attempt_history being private.

Should I move these tests to `crate::run::tests`, or is there something else I can do here?

Comment (CryZe) [reply src/timing/timer/tests/mod.rs]: https://docs.rs/livesplit-core/0.11.0/livesplit_core/run/struct.Run.html#method.attempt_history

Comment (CryZe) [reply src/timing/timer/tests/mod.rs]: Oh wait, you try to modify it. Uhm, then yeah I guess move it there

**Comment by TheEssenceOfDarkness** [comment]
This should be good now.

**Pull request merged**